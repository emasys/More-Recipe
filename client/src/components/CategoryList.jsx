import React from 'react';
import { Link } from 'react-router-dom';

import config from '../config/index';


export const list = [
  'Breakfast',
  'Brunch',
  'Lunch',
  'Snacks',
  'Appetisers',
  'Dinner',
  'Soups',
  'Noodles',
  'Rice',
  'Pasta',
  'Meat',
  'Poultry',
  'Seafood',
  'Vegetarian',
  'Sides',
  'Sauces',
  'Baking',
  'Desserts',
  'Drinks',
  'Salads'
];

const CategoryList = props => (
  <section className="mt-0 mb-20 text-center">
    <div className="category-wrapper container-fluid">
      <div className="containe">
        <div className="row justify-content-center">
          {list.map(link => (
            <div className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0" key={link}>
              <Link
                to={`/category/${link}`}
                className="categories hvr-shrink bg-dark"
              >
                <img
                  src={config.CAT_IMAGE[link]}
                  className="category-image"
                  alt="category"
                />
                <p>{link}</p>
              </Link>
            </div>
          ))}
        </div>
      </div>
    </div>
  </section>
);

export default CategoryList;
