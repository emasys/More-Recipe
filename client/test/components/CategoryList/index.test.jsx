import React from 'react';
import { shallow } from 'enzyme';

// Component
import CategoryList from '../../../src/components/CategoryList';

let wrapper = null;

describe('Test suite for Footer component', () => {
  beforeEach(() => {
    wrapper = shallow(<CategoryList />);
  });
  it('Should render without errors', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
