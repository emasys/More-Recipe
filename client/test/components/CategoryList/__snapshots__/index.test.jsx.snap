// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Footer component should match snapshot 1`] = `
<section
  className="mt-0 mb-20 text-center"
>
  <div
    className="category-wrapper container-fluid"
  >
    <div>
      <div
        className="row justify-content-center"
      >
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Breakfast"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Breakfast"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514747192/oatmeal-3-2-1-400x400_zjkinq.jpg"
            />
            <p>
              Breakfast
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Brunch"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Brunch"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514747533/Leftover-Stuffing-22Hash22-6-400x400_qsgvfy.jpg"
            />
            <p>
              Brunch
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Lunch"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Lunch"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514747756/Hotel_New_York_Rotterdam_Lunch_1-400x400_xwwedv.jpg"
            />
            <p>
              Lunch
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Snacks"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Snacks"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514747847/Oatmeal-Raisin-Cookie-Chex-Snack-Mix-400x400_grttrd.jpg"
            />
            <p>
              Snacks
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Appetisers"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Appetisers"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514747940/IMG_8476-400x400_iksx0h.jpg"
            />
            <p>
              Appetisers
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Dinner"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Dinner"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748067/chicken2-Medium-400x400_jdip45.jpg"
            />
            <p>
              Dinner
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Soups"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Soups"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748162/mushroom-soup-6-min-400x400_whcvep.jpg"
            />
            <p>
              Soups
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Noodles"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Noodles"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748289/stir-fried-chicken-and-cashews-with-cabbage-noodles-1780-400x400_gujnry.jpg"
            />
            <p>
              Noodles
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Rice"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Rice"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748439/Peas-Pulao-1_swucnu.png"
            />
            <p>
              Rice
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Pasta"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Pasta"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748566/Summertime-Pasta-4-400x400_v0xvsg.jpg"
            />
            <p>
              Pasta
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Meat"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Meat"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748713/Beef-or-lamb_dxs4l9.jpg"
            />
            <p>
              Meat
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Poultry"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Poultry"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748791/Gressingham-duck-400x400_vctzt9.jpg"
            />
            <p>
              Poultry
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Seafood"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Seafood"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514748878/seafood-appetizer-400x400_nkjeyc.jpg"
            />
            <p>
              Seafood
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Vegetarian"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Vegetarian"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514749041/veggie-bowl-recipe-fp-400x400-c_k4yjbi.jpg"
            />
            <p>
              Vegetarian
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Sides"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Sides"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514749108/10-sweet-potato-recipes-you-have-to-try-400x400_rgxzn2.jpg"
            />
            <p>
              Sides
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Sauces"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Sauces"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514749192/cocktail-sauce-recipe-fp-400x400-c_nsa9h3.jpg"
            />
            <p>
              Sauces
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Baking"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Baking"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514749250/paleo-vanilla-cake-recipe-fp-400x400-c_s4cimv.jpg"
            />
            <p>
              Baking
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Desserts"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Desserts"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514749309/angus-barn-chocolate-pie-04-400x400_hohb3e.jpg"
            />
            <p>
              Desserts
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Drinks"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Drinks"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514749359/mai-tai-blue-hawaiian-cocktails-recipe-fp-400x400-c_xfw2bc.jpg"
            />
            <p>
              Drinks
            </p>
          </Link>
        </div>
        <div
          className="col-lg-2 col-md-3 col-sm-4 mb-2 p-0"
          key="Salads"
        >
          <Link
            className="categories hvr-shrink bg-dark"
            replace={false}
            to="/category/Salads"
          >
            <img
              alt="category"
              className="category-image"
              src="https://res.cloudinary.com/emasys/image/upload/v1514749454/beet-salad-recipe-fp-400x400-c_d8zyu7.jpg"
            />
            <p>
              Salads
            </p>
          </Link>
        </div>
      </div>
    </div>
  </div>
</section>
`;
